src/Farmer/Types.fs

```diff
@@ -138,7 +138,7 @@ type ResourceId with
               this.Type.Type
               this.Name.Value
               for segment in this.Segments do segment.Value ]
-            |> List.map (fun x -> if x.StartsWith("[") then x.Trim('[' ,']') else sprintf $"'%s{x}'")
+            |> List.map (fun x -> if x.StartsWith("[") then x.Trim('[' ,']') else $"'%s{x}'") // Fixes case where a template e.g. [resourceGroup().name] is used.
```

Comment:
@JustinWilkinson I think this is correct?

Reply:
Yeah - this got merged in yesterday curtesy of Ninja Robot.